app/frontend/src/components/ConfirmationDialogWrapper.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { cloneElement, ReactElement, useState } from "react";
+
+import Button from "./Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "./Dialog";
+
+interface ConfirmationDialogWrapperProps<P extends { onClick: () => void }> {
+  children: ReactElement<P>;
+  title: string;
+  message: string;
+}
+
+export default function ConfirmationDialogWrapper<
+  P extends { onClick: () => void }
+>({ children, title, message }: ConfirmationDialogWrapperProps<P>) {
+  if (!children.props.onClick) {
+    throw new Error(
+      "Child of ConfirmationDialogWrapper must have onClick prop"
+    );
+  }
+  const [isOpen, setIsOpen] = useState(false);
+  const child = cloneElement<P>(children, {
+    onClick: () => setIsOpen(true),
+  } as P);
+  const ariaLabel = `${title.replace(/\s+/g, "")}-confirmation-dialog`;
+  const handleConfirm = () => {
+    children.props.onClick();
+    setIsOpen(false);
```

Comment:
This seems a bit unorthodox just to "hijack" the original children to get its `onClick` method, since the main plus of using the `children` prop is it's supposed to be generic/flexible.

I think it'd be easier and a bit clearer to pass through the confirmation action function as a prop like `onConfirm` and change `children` into a render prop function instead so `children` can remain generic while have access to the `setIsOpen` state setter.

Reply:
Yeah that's a better idea